### ISSUE TITLE
pipelines: Invoking a Lambda function

### ISSUE THREAD

**Issue by SydneyUni-Jim** [aws/aws-cdk#24978]
### Describe the feature

Add step that invokes a Lambda function before or after a stage.

### Use Case

While anything can be done with a `ShellStep`, the overhead of starting a CodeBuild project can be avoided by instead invoking a Lambda function.

An example is running an ECS task to apply database migrations or reconnecting another Lambda function to its SQS queue after passing in-situ testing.

### Proposed Solution

Add a new blueprint for a new `InvokeLambdaStep` following the modal of `ConfirmPermissionsBroadening`.

### Other Information

I'm existing capabilities would be able to build and deploy the Lambda function in the prior `Build` and `UpdatePipeline` stages.

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.73.0

### Environment details (OS name and version, etc.)

macos/13.3 (aarch64) nodejs/18.15.0 typescript/5.0.3

**Comment by pahud** [comment]
Yeah this would be very flexible! Thank you for your use case sharing.
